p2p/peer.py

```diff
@@ -865,6 +878,82 @@ def get_nodes_to_connect(self) -> Generator[Node, None, None]:
             yield node
 
 
+class PreferredNodePeerPool(PeerPool):
+    """
+    A PeerPool which has a list of preferred nodes which it will prioritize
+    using before going to the discovery protocol to find nodes.  Each preferred
+    node can only be used once every preferred_node_recycle_time seconds.
+    """
+    preferred_nodes: Sequence[Node] = None
+    preferred_node_recycle_time: int = 300
+    _preferred_node_tracker: Dict[Node, float] = None
```

Comment:
Not as part of this PR but would it make sense to maintain a list of preferred nodes in the db or in some updated configuration file so that the list of preferred nodes becomes something well maintained (with newly discovered peers added and stale peers removed) to not fall back to the hardcoded nodes whenever there's a restart?

Reply:
:+1: https://github.com/ethereum/py-evm/issues/757